src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
```

Comment:
agrrrhhh!!! :) now I changed checking to Iterator. You can fix it later as you wish :)